mlem/cli/declare.py

```diff
@@ -38,3 +45,83 @@ def declare(
     with wrap_build_error(subtype, cls):
         meta = build_mlem_object(cls, subtype, conf, [])
     meta.dump(path, project=project, index=index, external=external)
+
+
+declare2 = Typer(
+    name="declare2",
+    help="Manage deployments",
+    cls=mlem_group("objects", aliases=["asdasd"]),
+)
+app.add_typer(declare2)
+
+
+def get_field_help(cls: Type, field_name: str):
+    return (
+        get_attribute_docstring(cls, field_name).docstring_below
+        or "Field docstring missing"
+    )
+
+
+def create_declare(typename, cls: Type[MlemObject]):
+    if cls.__is_root__:
+        typer = Typer(name=typename, help=cls.__doc__, cls=mlem_group("lal"))
+        declare2.add_typer(typer)
+
+        for subtype in list_implementations(MlemObject, cls):
+            _how_should_i_name_this(subtype, typename, typer)
+    else:
+
+        @mlem_command(typename, parent=declare2)
+        def subtype_command(
+            path: str = Argument(..., help="Where to save object"),
+            project: str = option_project,
+            external: bool = option_external,
+            index: bool = option_index,
+        ):
+            cls = MlemObject.__type_map__[subtype]
+            with wrap_build_error(subtype, cls):
+                meta = build_mlem_object(cls, subtype, [], [])
+            meta.dump(path, project=project, index=index, external=external)
+
+
+def _how_should_i_name_this(subtype, typename, typer):
+    subtype_cls = load_impl_ext(typename, subtype)
+
+    class SuperduperMlemCommand(MlemCommand):
+        def get_help(self, ctx) -> str:
+            self._getting_help = True
+            return super().get_help(ctx)
+
+        def get_params(self, ctx) -> List["Parameter"]:
+            res: List[Parameter] = super().get_params(ctx)
+            if hasattr(self, "_getting_help"):
+                c_param = [p for p in res if p.name == "conf"][0]
+                res.remove(c_param)
+                for arg in subtype_cls.__fields__:
+                    arg_param = copy.copy(c_param)
+                    arg_param.secondary_opts = arg_param.secondary_opts + [arg]
+                    arg_param.help = get_field_help(subtype_cls, arg)
+                    res.append(arg_param)
+            return res
+
+    @mlem_command(subtype, parent=typer, mlem_cls=SuperduperMlemCommand)
+    def subtype_command(
+        path: str = Argument(..., help="Where to save object"),
+        project: str = option_project,
+        external: bool = option_external,
+        index: bool = option_index,
+        conf: Optional[List[str]] = Option(
+            None,
+            "-c",
+            "--conf",
+            help="Values for object fields in format `field.nested.name=value`",
```

Comment:
I'm not sure I'm getting this `mlem declare2` syntax TBH. Maybe a bit more explanation or quick demo in the PR description?

In any case, if we follow the DVC pattern perhaps we need a `mlem config` command instead and require 2 steps to fully set up the different objects and other resources e.g. something like

```console
$ mlem create env heroku staging
$ mlem config env staging --api_key=<you api key>
```

Reply:
I dont see why we need to follow DVC pattern, I mean we already have `mlem config` (for mlem configuration) and why 2 steps are better than one?
For now we can read secrets from env vars, so you don't actually need to set them. We might add something like `--local` in the future, but it is out of scope ow this PR/issue.
As for using `--<var name> <value>`, I will try to add this, possibly it will be even cleaner code. But I'd keep `-c` option just in case of weird var names

Reply:
> why we need to follow DVC pattern

That's not the intention, but if DVC does it that way there's probably a reason. And UX consistency across Iterative tools may be desirable.

I see this is not the right PR to discuss that though 🙂